src/frame/plane.rs

```diff
@@ -787,7 +793,55 @@ pub mod test {
       &downsampled.data[..]
     );
   }
-
+  #[test]
+  fn test_plane_downsample_odd() {
+    #[rustfmt::skip]
+    let plane = Plane::<u8> {
+      data: PlaneData::from_slice(&[
+        0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 1, 2, 3, 4, 0, 0,
+        0, 0, 8, 7, 6, 5, 0, 0,
+        0, 0, 9, 8, 7, 6, 0, 0,
+        0, 0, 2, 3, 4, 5, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0,
+      ]),
+      cfg: PlaneConfig {
+        stride: 8,
+        alloc_height: 9,
+        width: 5,
+        height: 5,
+        xdec: 0,
+        ydec: 0,
+        xpad: 0,
+        ypad: 0,
+        xorigin: 2,
+        yorigin: 3,
+      },
+    };
+    let downsampled = plane.downsampled(5, 5);
+    
+    #[rustfmt::skip]
+    assert_eq!(
+      &[
+        5, 5, 0, 0, 0, 0, 0, 0,  
+        0, 0, 0, 0, 0, 0, 0, 0,  
+        0, 0, 0, 0, 0, 0, 0, 0,  
+        0, 0, 0, 0, 0, 0, 0, 0,  
+        6, 6, 0, 0, 0, 0, 0, 0,  
+        0, 0, 0, 0, 0, 0, 0, 0,  
+        0, 0, 0, 0, 0, 0, 0, 0,  
+        0, 0, 0, 0, 0, 0, 0, 0,  
+        0, 0, 0, 0, 0, 0, 0, 0,  
+        0, 0, 0, 0, 0, 0, 0, 0,  
+        0, 0, 0, 0, 0, 0, 0, 0,  
+        0, 0, 0, 0, 0, 0, 0, 0,  
```

Comment:
How about changing the width and height to 3 from 5 and expected result as same as
test_plane_downsample() as L784-791?

Reply:
Yes, very much we can have that,

I thought of having higher values for both width and height, the only issue was when we have large values for Width/Height like 9,11,13 it would crash. 
IIRC we have a hard limit of 16x16 for width and height
Ref: https://github.com/xiph/rav1e/blob/master/src/api/config.rs#L774
I am not sure what we are missing.

I did update the PR too, please do check it

Reply:
> I thought of having higher values for both width and height, the only issue was when we have large values for Width/Height like 9,11,13 it would crash.
For this, we need to change the stride and alloc_height too.
I have confirmed Width/Height=9 works without crash by changing stride and alloc_height.

For testing, I believe Width/Height=3 is enough.